edge/pkg/edged/edged.go

```diff
@@ -220,20 +225,44 @@ func newEdged(enable bool, nodeName, namespace string) (*edged, error) {
 	return ed, nil
 }
 
-func (e *edged) syncPod(podCfg *config.PodConfig) {
-	time.Sleep(10 * time.Second)
+func (e *edged) syncPod(podCfg *config.PodConfig, kubeletErrChan <-chan error) {
+	startWaiter := time.NewTimer(10 * time.Second)
```

Comment:
Could u plz explain more about the modification of Line 229-249?

Reply:
If we use `time.Sleep`, when the goroutine which runs kubelet failed without calling `os.Exit(1)`, `syncPod` cannot returns immediately.

To make `syncPod` return once kubelet fails, use a Timer `startWaiter` to provide a channel instead of just sleep the goroutine.

With the channel `startWaiter.C`, it covers three conditions:
 
1. global context exits, `syncPod` must return
2. kubelet fails, `syncPod` must return
3. `syncPod` has waited for 10s, `syncPod` is ready to sync pods

I've misunderstood godoc for timer and removed the unnecessary draining channel operation here.